### PR TITLE
Support generate document automatically for pulsar-client

### PR DESCRIPTION
**Pull request by 315157973**
Master Issue: #10040

### Modifications

add command that can  generate document automatically for pulsar-client

### CODE REVIEW

File: pulsar-client-tools/src/main/java/org/apache/pulsar/client/cli/PulsarClientTool.java [hunk 1]
```diff
@@ -110,14 +111,15 @@ public PulsarClientTool(Properties properties) {
...
+        generateDocument = new CmdGenerateDocument();
 
         this.commandParser = new JCommander();
         this.usageFormatter = new DefaultUsageFormatter(this.commandParser);
-
         commandParser.setProgramName("pulsar-client");
         commandParser.addObject(this);
         commandParser.addCommand("produce", produceCommand);
         commandParser.addCommand("consume", consumeCommand);
+        commandParser.addCommand("document", generateDocument);
```
Comment (eolivelli) [inline pulsar-client-tools/src/main/java/org/apache/pulsar/client/cli/PulsarClientTool.java:122]: What about 'generate_documentation ' ?

**Review by eolivelli — COMMENTED** [review]
Nice work.

I guess this command is intended to be execured inside some automated workflow while generating the website.

Or what's the plan?

File: pulsar-client-tools/src/main/java/org/apache/pulsar/client/cli/PulsarClientTool.java (hunk 1 above)
Comment (315157973) [reply pulsar-client-tools/src/main/java/org/apache/pulsar/client/cli/PulsarClientTool.java:122]: Good idea, I will revise it.
Actually, I don’t know how to integrate it into the web page. But now there must be automated tools: http://pulsar.apache.org/tools/pulsar-admin/2.7.0-SNAPSHOT/
cc @codelipenghui

File: pulsar-client-tools/src/main/java/org/apache/pulsar/client/cli/CmdGenerateDocument.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+@Getter
+@Parameters(commandDescription = "Generate documents automatically.")
+@Slf4j
+public class CmdGenerateDocument {
```
Comment (merlimat) [inline pulsar-client-tools/src/main/java/org/apache/pulsar/client/cli/CmdGenerateDocument.java:38]: --> `CmdGenerateDocumentation`

**Comment by Anonymitaet** [comment]
@315157973 thanks for your great work! 
@tuteng could you please help review? Thanks

**Comment by tuteng** [comment]
@315157973 Thanks, great work, local test, works fine, we can merge this pr first, then I will add an automated workflow

image

**Review by tuteng — APPROVED** [review]
LGTM

**Review by eolivelli — APPROVED** [review]

**Comment by Anonymitaet** [comment]
@tuteng many thanks for your guidance! Please ping me once you finish the work, then we can help review and check the contents on the website.

**Pull request merged**